scertecd/scertecd.go

```diff
@@ -643,14 +672,20 @@ func (s *Server) parseCertMeta(p []byte) (*certMeta, error) {
 	if !strings.HasSuffix(blocks[0].Type, " PRIVATE KEY") {
 		return nil, errors.New("first PEM block is not a private key")
 	}
-	for _, cb := range blocks[1:] {
+	certBlocks := blocks[1:]
+	for i, cb := range certBlocks {
 		if cb.Type != "CERTIFICATE" {
 			return nil, fmt.Errorf("unexpected PEM block of type %q", cb.Type)
 		}
 		c, err := x509.ParseCertificate(cb.Bytes)
 		if err != nil {
 			return nil, fmt.Errorf("parsing cert: %w", err)
 		}
+		if i == 0 {
+			m.Leaf = c
+		} else {
+			m.Issuer = c
```

Comment:
Does it matter if there are > 2 remaining after the private key, that is, do we know the issuer is always "last"?

Reply:
Yes, it's always last AFAIU. But there's also only 2 these days. I kinda doubt LetsEncrypt is going to add an intermediate back. Maybe I should just lock down all the code paths to enforce exactly 2.